tests/test_vert_coords.py

```diff
@@ -0,0 +1,125 @@
+
+    assert type(result) == type(ptop)
+    if isinstance(result, np.ndarray):
+        assert len(result) == len(ptop)
+
+
+def test_atmosphere_hybrid_sigma_pressure_coordinate_to_pressure_exceptions():
+    with pytest.raises(ValueError):
+        atmosphere_hybrid_sigma_pressure_coordinate_to_pressure(np.ones(10), np.ones(5), 1, 0)
+
+
+@pytest.mark.parametrize(
+    "a,b,ps,p0",
+    (
+        pytest.param(np.ones(10), np.ones(10), 1, None, id="test-p0-is-none"),
+        pytest.param(np.ones(10), np.ones(10), 1, 1, id="test-p0-is-not-none"),
+    ),
+)
+def test_atmosphere_hybrid_sigma_pressure_coordinate_to_pressure(
+    a: np.ndarray, b: np.ndarray, ps: float, p0: float | None
+):
+    result = atmosphere_hybrid_sigma_pressure_coordinate_to_pressure(a, b, ps, p0)
+
+    assert len(a) == len(b) == len(result)
+
+
+def test_geopotentialheight2altitude():
+    with pytest.raises(NotImplementedError):
+        geopotentialheight2altitude(5)
+
+
+def test_VerticalCoordinate_exceptions():
+    # Unsupported variable name during initialization.
+    with pytest.raises(ValueError):
+        VerticalCoordinate("jtlkjhsklrg")
+
+    # Attempt calculating pressure for unsupported variable.
+    vert = VerticalCoordinate("asc")
+    with pytest.raises(CoordinateNameError):
+        vert.calc_pressure(np.ones(5))
+
+
+@pytest.mark.parametrize(
+    "name,expected,exception",
+    (
+        pytest.param("altitude", True, None, id="var-name-1"),
+        pytest.param("atmosphere_sigma_coordinate", False, None, id="var-name-2"),
+        pytest.param("atmosphere_hybrid_sigma_pressure_coordinate", False, None, id="var-name-3"),
+        pytest.param("asc", False, None, id="standard-name-1"),
+        pytest.param("z", True, None, id="standard-name-2"),
+        pytest.param("gph", None, ValueError, id="valueerror"),
+    ),
+)
+def test_VerticalCoordinate_lev_increases_with_alt(
+    name: str, expected: bool | None, exception: Exception | None
+):
+    vert = VerticalCoordinate(name)
+    if exception:
+        with pytest.raises(exception):
+            vert.lev_increases_with_alt
+    else:
+        assert vert.lev_increases_with_alt == expected
+
+
+def test_AltitudeAccess_exceptions():
+    with pytest.raises(ValueError):
+        AltitudeAccess(None)
+
+
+def test_AltitudeAccess_get_altitude(alt: AltitudeAccess):
+    with pytest.raises(NotImplementedError):
+        alt.get_altitude(None, None)
+
+
+@lustre_unavail
+@pytest.mark.xfail
+# TODO: This test is currently failing because of a ValueError caused by
+# a variable name containing an underscore. This seems like an unrelated
+# issue to me, so I have marked it xfail for now.
+def test_AltitudeAccess_check_access(alt: AltitudeAccess):
+    assert alt.check_altitude_access()
+
+
+def test_AltitudeAccess_search_aux_coords(alt: AltitudeAccess):
+    with pytest.raises(CoordinateNameError):
+        alt.search_aux_coords(["gkjfdshglk"])
```

Comment:
please put the exception part on a separate test

Reply:
Done.